### PR TITLE
Feature/boolean filter menu

### LINKED ISSUE #210: Filter box for boolean dimension is too complex
Filter for boolean dimension could be simplified but it must handle ternary logic (true/false/null).

### PR DESCRIPTION
**Pull request by adrianmroz**
Closes #210 #252

### CODE REVIEW

**Comment by adrianmroz** [comment]
https://github.com/allegro/turnilo/blob/feature/boolean-filter-menu/src/common/utils/formatter/formatter.ts#L94

Handle all cases with switch and include Boolean filter clause explicitly. Cast value to string (booleans in JSX are omitted)

File: src/common/utils/formatter/formatter.ts [hunk 1]
```diff
@@ -91,18 +92,29 @@ function getFormattedStringClauseValues({ values, action }: StringFilterClause):
   }
 }
 
+function getFormattedBooleanClauseValues({ values }: BooleanFilterClause) {
```
Comment (mkuthan) [inline src/common/utils/formatter/formatter.ts]: no return type

File: src/common/utils/formatter/formatter.ts [hunk 2]
```diff
@@ -91,18 +92,29 @@ function getFormattedStringClauseValues({ values, action }: StringFilterClause):
...
+  return values.count() > 1 ? `(${values.count()})` : values.first().toString();
+}
+
+function getFormattedNumberClauseValues(clause: NumberFilterClause) {
```
Comment (mkuthan) [inline src/common/utils/formatter/formatter.ts]: no return type

File: src/common/utils/formatter/formatter.ts [hunk 3]
```diff
@@ -91,18 +92,29 @@ function getFormattedStringClauseValues({ values, action }: StringFilterClause):
...
+  const { start, end } = clause.values.first();
+  return `${start} to ${end}`;
+}
+
 function getFilterClauseValues(clause: FilterClause, timezone: Timezone): string {
```
Comment (mkuthan) [inline src/common/utils/formatter/formatter.ts]: Could we use polymorphic calls here if the  move getFormattedTimeClauseValues to the clause class? But how to handle timezone for time clause?

File: src/client/components/boolean-filter-menu/boolean-filter-menu.tsx [hunk 4]
```diff
@@ -0,0 +1,183 @@
...
+  }
+
+  componentDidMount() {
+    this.fetchData();
+  }
+
+  fetchData() {
+    const { essence, timekeeper, dimension } = this.props;
+    const { dataCube } = essence;
+    const $main = $("main");
```
[the remark this answers is not in the document]
Comment (adrianmroz) [reply src/client/components/boolean-filter-menu/boolean-filter-menu.tsx]: One "fix code" too far. I'll just inline that.

File: src/client/components/filter-menu/filter-menu.tsx [hunk 5]
```diff
@@ -23,6 +23,7 @@ import { Essence } from "../../../common/models/essence/essence";
 import { Stage } from "../../../common/models/stage/stage";
 import { Timekeeper } from "../../../common/models/timekeeper/timekeeper";
 import { Fn } from "../../../common/utils/general/general";
+import { BooleanFilterMenu } from "../boolean-filter-menu/boolean-filter-menu";
```
[the remark this answers is not in the document]
Comment (adrianmroz) [reply src/client/components/filter-menu/filter-menu.tsx]: I would go for 
```
filter-menu
  - filter-menu.tsx
  - boolean-menu
    - boolean-filter-menu.tsx
```

Subdirectories to keep styles and test files in check.

File: src/common/utils/formatter/formatter.ts (hunk 3 above)
Comment (adrianmroz) [reply src/common/utils/formatter/formatter.ts]: In perfect case I would define `getFilterClauseValues` as trait/typeclass and define implementations outside of `FilterClause`. There is a lot of "formatter" logic inside and I think it belongs there.

**Pull request merged**